barretenberg/cpp/pil/vm2/opcodes/internal_call.pil

```diff
@@ -17,7 +17,9 @@ namespace execution;
     // When we encounter this case, the internal call information in the next row is constrained to change (incremented, unwound, etc)
     pol NEW_NEXT_CALL_ID = (sel_execute_internal_call + sel_execute_internal_return) * (1 - sel_error) + sel_exit_call;
 
-    pol PROPAGATE_CALL_ID = 1 - RESET_NEXT_CALL_ID - NEW_NEXT_CALL_ID;
+    // This is an XOR
+    pol RESET_OR_NEW_NEXT_CALL_ID = (RESET_NEXT_CALL_ID + NEW_NEXT_CALL_ID) - (RESET_NEXT_CALL_ID * NEW_NEXT_CALL_ID);
+    pol PROPAGATE_CALL_ID = 1 - RESET_OR_NEW_NEXT_CALL_ID;
```

Comment:
From [here](https://github.com/AztecProtocol/aztec-packages/pull/16428/files#diff-8b88bbe2fe99141bbb625e605c2e2c4eb0f8d948c49ad76fe16d6891de8d4a36R128), `RESET_NEXT_CALL_ID` and `NEW_NEXT_CALL_ID` weren't mutually exclusive.